src/js/components/table.js

```diff
@@ -0,0 +1,230 @@
+    )(getCellValue(direction ? a : b, index), getCellValue(direction ? b : a, index));
+
+/**
+ * Get an Array of column headers elements belonging directly to the given
+ * table element.
+ * @param {HTMLTableElement} table
+ * @return {array<HTMLTableHeaderCellElement>}
+ */
+const getColumnHeaders = (table) => {
+  const headers = select(SORTABLE_HEADER, table);
+  return headers.filter((header) => header.closest(TABLE) === table);
+};
+
+/**
+ * Update the button label within the given header element, resetting it
+ * to the default state (ready to sort ascending) if it's no longer sorted
+ * @param {HTMLTableHeaderCellElement} header
+ */
+const updateSortLabel = (header) => {
+  const headerName = header.innerText;
+  const sortedAscending = header.getAttribute(SORTED) === ASCENDING;
+  const isSorted = header.getAttribute(SORTED) === ASCENDING || header.getAttribute(SORTED) === DESCENDING || false;
+  const headerLabel = `${headerName}', sortable column, currently ${isSorted ? `${sortedAscending ? `sorted ${ASCENDING}` : `sorted ${DESCENDING}`}`: 'unsorted'}`;
+  const headerButtonLabel = `Click to sort by ${headerName} in ${sortedAscending ? DESCENDING : ASCENDING} order.`;
+  header.setAttribute("aria-label", headerLabel);
+  header.querySelector(SORT_BUTTON).setAttribute("title", headerButtonLabel);
+}
+
+/**
+ * Remove the aria-sort attribute on the given header element, and reset the label and button icon
+ * @param {HTMLTableHeaderCellElement} header
+ */
+const unsetSort = (header) => {
+  header.removeAttribute(SORTED);
+  updateSortLabel(header);
+}
+
+/**
+ * Sort rows either ascending or descending, based on a given header's aria-sort attribute
+ * @param {HTMLTableHeaderCellElement} header
+ * @param {boolean} ascending
+ * @return {boolean} true
+ */
+const sortRows = (header, ascending) => {
+  header.setAttribute(SORTED, ascending === true ? DESCENDING : ASCENDING );
+  updateSortLabel(header);
+
+  const tbody = header.closest(TABLE).querySelector('tbody');
+
+  // We can use Array.from() and Array.sort() instead once we drop IE11 support, likely in the summer of 2021
+  //
+  // Array.from(tbody.querySelectorAll('tr').sort(
+  //   compareFunction(
+  //     Array.from(header.parentNode.children).indexOf(header),
+  //     !ascending)
+  //   )
+  // .forEach(tr => tbody.appendChild(tr) );
+
+  // [].slice.call() turns array-like sets into true arrays so that we can sort them
+  const allRows = [].slice.call(tbody.querySelectorAll('tr'));
+  const allHeaders = [].slice.call(header.parentNode.children);
+  const thisHeaderIndex = allHeaders.indexOf(header);
+  allRows.sort(
+    compareFunction(
+      thisHeaderIndex,
+      !ascending
+    )
+  ).forEach(tr => {
+    [].slice.call(tr.children).forEach(td => td.removeAttribute("data-sort-active"));
```

Comment:
Would `[...tr.children]` also work in this case?

Does babel not cover either of those cases?

Reply:
Unfortunately not. In IE11, this fails with "Invalid attempt to spread non-iterable instance"